### ISSUE TITLE
result of DSA doesn't look like a keycap, Warnings: "Ignoring unknown variable '$side_sculpting_factor'" (workaround discovered)

### ISSUE THREAD

**Issue by czyz** [rsheldiii/KeyV2#126]
The results of trying to produce a DSA keycap using this modified version of the "// example key" line:

`dsa_row(3) legend("D", size=9) key();
`
<img width="371" alt="Screen Shot 2022-01-18 at 16 22 49" src="https://user-images.githubusercontent.com/345036/150040194-4bc2dce1-86b4-4ce5-b5c4-c5ee17d31f28.png">

In the process, the OpenSCAD console fills with tons of messages that look like the following:
`WARNING: Ignoring unknown variable '$side_sculpting_factor' in file src/shapes/sculpted_square.scad, line 13
`

I took a look at sculpted_square.scad but nothing jumped out at me as wrong -- but I am inexperienced with this software.

Tried with both "2022.01.17 (git d0be100cd)" and 2022.01.12 on macOS 11.6.

**Comment by czyz** [comment]
I run into the same issue with ds, dss, sa, hipro.

I took a look at the way variables are set up in rounded_square.scad compared to sculpted_square.scad, and only sculpted_square.scad included variable assignments before and outside its "module", so I thought maybe for whatever reason these variables are just not getting assigned. I copied them into keys.scad above the line that creates my key, and after that keys.scad will compile and generate ds, dss, sa, hipro, and dsa keycaps without errors.

I lack the OpenSCAD knowledge to fix sculpted_square.scad such that these variable assignments work properly when they're declared within that file, but by setting up these 3 variables in keys.scad I can now generate dsa keycaps, so I'm proceeding with this workaround for now.

`include <./includes.scad>


// variables taken from rounded_square.scad

// When sculpting sides, how much in should the tops come
$side_sculpting_factor = 4.5;
// When sculpting corners, how much extra radius should be added
$corner_sculpting_factor = 1;
// When doing more side sculpting corners, how much extra radius should be added
$more_side_sculpting_factor = 0.4;


// example key
dsa_row(1) legend("2", size=9) key();
`
